java/ql/src/semmle/code/java/dataflow/internal/DataFlowImpl.qll

```diff
@@ -1728,37 +1732,42 @@ private class PathNodeSink extends PathNode, TPathNodeSink {
  * a callable is recorded by `cc`.
  */
 private predicate pathStep(PathNodeMid mid, Node node, CallContext cc, AccessPath ap) {
-  localFlowBigStep(mid.getNode(), node, true, mid.getConfiguration()) and
-  cc = mid.getCallContext() and
-  ap = mid.getAp()
-  or
-  localFlowBigStep(mid.getNode(), node, false, mid.getConfiguration()) and
-  cc = mid.getCallContext() and
-  mid.getAp() instanceof AccessPathNil and
-  ap = node.(AccessPathNilNode).getAp()
-  or
-  jumpStep(mid.getNode(), node, mid.getConfiguration()) and
-  cc instanceof CallContextAny and
-  ap = mid.getAp()
-  or
-  additionalJumpStep(mid.getNode(), node, mid.getConfiguration()) and
-  cc instanceof CallContextAny and
-  mid.getAp() instanceof AccessPathNil and
-  ap = node.(AccessPathNilNode).getAp()
-  or
-  contentReadStep(mid, node, ap) and cc = mid.getCallContext()
-  or
-  exists(Content f, AccessPath ap0 | contentStoreStep(mid, node, ap0, f, cc) and push(ap0, f, ap))
-  or
-  pathOutOfArgument(mid, node, cc) and ap = mid.getAp()
-  or
-  pathIntoCallable(mid, node, _, cc, _) and ap = mid.getAp()
-  or
-  pathOutOfCallable(mid, node, cc) and ap = mid.getAp()
-  or
-  pathThroughCallable(mid, node, cc, ap)
+  exists(LocalCallContext localCC | localCC.matchesCallContext(cc) |
```

Comment:
Merely going from a `CallContext` to a `LocalCallContext` might risk pruning all local flow from certain callables if a call site can dispatch to two targets and is relevant for local flow pruning in one target and relevant for dispatch in the other.  We need to take the current callable into consideration when translating.